### ISSUE TITLE
H2Load Following Redirects

### ISSUE THREAD

**Issue by forbesb** [nghttp2/nghttp2#1140]
Is there a plan to implement HTTP redirects (3xx status codes) in h2load?
Currently it returns the number of redirects, but does not follow them.

**Comment by LPardue** [comment]
I don't think this is impossible but I do think this would be difficult with the way h2load is designed.

Request objects are created in an early step before testing commences. Once testing begins the connection objects (TCP socket + TLS) are created and assigned Requests (more connections may get made if there is ramp-up behavior in the test).

Following redirects would add more Request objects that need to be created on-the-fly as a result of server Response. I'm not sure the request-to-connection mapping logic would accomodate that in its current form. 

Redirects would also need h2load to parse the response, which it does sparingly. With that you'd probably want some infinite-loop detection and/or redirect count limit.

Then, the final statistics would need to be modified in a way that made sure they were coherent if/when redirects occurred and the client followed them.
